pkg/vpcmodel/subnetsConnectivity.go

```diff
@@ -252,14 +255,45 @@ func (v *VPCsubnetConnectivity) computeAllowedConnsCombined() error {
 			v.AllowedConnsCombined[src][dst] = combinedConns
 		}
 	}
+	return nil
+}
 
+func (v *VPCsubnetConnectivity) computeStatefulConnections() error {
+	for src, endpointConns := range v.AllowedConnsCombined {
+		for dst, conns := range endpointConns {
+			if conns.IsEmpty() {
+				continue
+			}
+			dstObj := v.cloudConfig.NameToResource[dst.Name()]
+			var otherDirectionConn *common.ConnectionSet = nil
+			switch dstObj.(type) {
+			case NodeSet:
+				otherDirectionConn = v.AllowedConnsCombined[dst][src]
+			case *ExternalNetwork:
+				// subnet to external node is stateful if the subnet's nacl allows egress from that node
+				otherDirectionConn = v.AllowedConns[src].EgressAllowedConns[dst]
```

Comment:
I do not understand this line: from the lines above, we know that there is allowed conn `src => dst : conns`, and in this case of the switch, `dst` is external network. 
So for it to be considered `stateful`, the subnet `src` should be able to receive responses from this external network, so there should be an allowed connection  from `dst` to `src`. why checking `EgressAllowedConns` to `dst`? 
Why not checking there is an allowed combined connection from `dst` to `src`?

Reply:
Because there is no actual connection in this direction, only for stateful. Hence this direction will not be part of the combined connections.

Reply:
Fixed to Ingress, of course